File: code/insec/receiver.py
import socket
import os


def start_udp_listener():
    # Create a UDP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    # Bind the socket to the port
    server_address = ('', 8888)
    sock.bind(server_address)

    print("UDP listener started on port 8888")

    while True:
        data, address = sock.recvfrom(4096)
        print(f"Received {len(data)} bytes from {address}")
        print(data.decode())
        data = "Hi SecureNet!".encode()
        if data:
            sent = sock.sendto(data, address)
            print(f"Sent {sent} bytes back to {address}")


def start_tcp_listener():
    # Create a TCP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Bind the socket to the port
    server_address = ('', 8888)
    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
    sock.bind(server_address)

    # Listen for incoming connections
    sock.listen()

    print("TCP listener started on port 8888")

    while True:
        connection, client_address = sock.accept()
        try:
            print(f"Connection from {client_address}")
            while True:
                data = connection.recv(4096)
                if data:
                    print(f"Received {len(data)} bytes from {client_address}")
                    print(data.decode())
                    connection.sendall("Hi SecureNet!".encode())
                else:
                    break
        finally:
            connection.close()


if __name__ == "__main__":
    start_tcp_listener()

File: code/insec/receiver.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

#define PORT 8888
#define BUFFER_SIZE 1024

int main() {
    int sockfd;
    char buffer[BUFFER_SIZE];
    struct sockaddr_in server_addr, client_addr;
    socklen_t addr_len = sizeof(client_addr);
    ssize_t recv_len;

    // Create socket
    if ((sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
        perror("Socket creation failed");
        exit(EXIT_FAILURE);
    }

    // Fill server information
    memset(&server_addr, 0, sizeof(server_addr));
    server_addr.sin_family = AF_INET;
    server_addr.sin_addr.s_addr = INADDR_ANY;
    server_addr.sin_port = htons(PORT);

    // Bind the socket with the server address
    if (bind(sockfd, (const struct sockaddr *)&server_addr, sizeof(server_addr)) < 0) {
        perror("Bind failed");
        close(sockfd);
        exit(EXIT_FAILURE);
    }

    while (1) {
        printf("Waiting for data...\n");
        recv_len = recvfrom(sockfd, buffer, BUFFER_SIZE, 0, (struct sockaddr *)&client_addr, &addr_len);
        if (recv_len < 0) {
            perror("Receive failed");
            continue;
        }

        buffer[recv_len] = '\0';
        printf("Received: %s\n", buffer);

        // Send response
        const char *response = "Message received";
        sendto(sockfd, response, strlen(response), 0, (struct sockaddr *)&client_addr, addr_len);
        printf("Response sent\n");
    }

    close(sockfd);
    return 0;
}